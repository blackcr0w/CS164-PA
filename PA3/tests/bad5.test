class Test {
  foo:Test;
  bar():Int {5};
};

(* error in wrong class definition *)
class Test extends Object {
  foo:Test;
};

class Test {
  foo:Test;
  bar():Int {5};
};

(* error in wrong class deifinition *)
class Test implements Nothing{
  foo:Test;
  bar():Int {5};
};


