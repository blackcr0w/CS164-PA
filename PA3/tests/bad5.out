#name "PA3/tests/bad5.test"
#1 CLASS
#1 OBJECTID a
#1 '+'
#1 OBJECTID b
#1 '{'
#2 OBJECTID emem
#3 '}'
#3 ';'
#5 CLASS
#5 OBJECTID c
#5 '+'
#5 OBJECTID d
#5 '{'
#6 CLASS
#6 OBJECTID a
#6 ':'
#7 CLASS
#7 OBJECTID b
#7 '('
#7 ')'
#7 ';'
#8 '}'
#8 ';'